pkg/executor/batch_point_get.go

```diff
@@ -236,7 +221,26 @@ func (e *BatchPointGetExec) initialize(ctx context.Context) error {
 	var handleVals map[string][]byte
 	var indexKeys []kv.Key
 	var err error
-	batchGetter := e.batchGetter
+	var batchGetter kv.BatchGetter = e.snapshot
+	sessVars := e.Ctx().GetSessionVars()
+	txnCtx := sessVars.TxnCtx
+	txn, err := e.Ctx().Txn(false)
+	if err != nil {
+		return err
+	}
+	if txn.Valid() {
+		lock := e.tblInfo.Lock
+		snapshot := e.Ctx().GetSessionVars().StmtCtx.MemBufferSnapshot
+		println(fmt.Sprintf("DBG Build batch get for snapshot, %v", snapshot))
+		if e.lock {
+			batchGetter = driver.NewBufferSnapshotBatchGetter(snapshot, &PessimisticLockCacheGetter{txnCtx: txnCtx}, e.snapshot)
+		} else if lock != nil && (lock.Tp == ast.TableLockRead || lock.Tp == ast.TableLockReadOnly) && e.Ctx().GetSessionVars().EnablePointGetCache {
+			batchGetter = newCacheBatchGetter(e.Ctx(), e.tblInfo.ID, e.snapshot)
+		} else if snapshot != nil {
+			batchGetter = driver.NewBufferSnapshotBatchGetter(snapshot, nil, e.snapshot)
```

Comment:
It could be confusing to use `snapshot` with `e.snapshot`, maybe rename the `snapshot` to `memBufferSnapshot`.